libbeat/keystore/keystore.go

```diff
@@ -34,6 +34,8 @@ var (
 	ErrKeyDoesntExists = errors.New("cannot retrieve the key")
 )
 
+var CentralKeystore Keystore
```

Comment:
this is a hack. We should avoid global variables as they can cause issues when reusing code, among many other problems (I'm ok with this for testing the approach, but we need to find another solution before merging). 

I think the `Beat` object is exposing the keystore through `beat.Keystore()` method. Probably you can get it somewhere and pass it to autodiscover.